src/TimeService.Business/Commands/Stat/FindStatCommand.cs

```diff
@@ -103,6 +109,59 @@ private async Task<List<ProjectData>> GetProjectsThroughBroker(List<Guid> projec
       return null;
     }
 
+    private async Task<List<UserData>> GetUsersData(List<Guid> usersIds, List<string> errors)
+    {
+      if (usersIds == null || !usersIds.Any())
+      {
+        return null;
+      }
+
+      List<UserData> usersFromCache = await _redisHelper.GetAsync<List<UserData>>(Cache.Users, usersIds.GetRedisCacheHashCode());
+
+      if (usersFromCache != null)
+      {
+        _logger.LogInformation("UserDatas were taken from the cache. Users ids: {usersIds}", string.Join(", ", usersIds));
+
+        return usersFromCache;
+      }
+
+      return await GetUsersDataFromBrokerAsync(usersIds, errors);
+    }
+
+    private async Task<List<UserData>> GetUsersDataFromBrokerAsync(List<Guid> usersIds, List<string> errors)
+    {
+      if (usersIds == null || !usersIds.Any())
+      {
+        return null;
+      }
+
+      string message = "Cannot get users data. Please try again later.";
```

Comment:
What for do you create the message here, If you use it only 160 string?

Reply:
fixed